### PR TITLE
[ci] fix: fix fsdp test in transformers 4.54.1

### PR DESCRIPTION
**Pull request by vermouth1992**
### What does this PR do?

When we upgrade to transformers 4.54.1, the fsdp checkpoint manager test breaks, and here are some observations:
- If we switch the "attn_implementation" to "eager" or "sdpa", everything works fine. So, it suggests that the issue lies within the flash_attention_2 backend of transformers.
- Previously, this test passes in input_ids and attention_mask. However, workers in verl passes in input_ids and position_ids to utilize rmpad. After we switch the input to `input_ids` and `position_ids`, all the tests passed.
- If we do not call loss.backward, everything works fine
- So, FSDP works fine, checkpoint manager works fine. The problem must lie in how transformers handles different type of input combinations in flash_attention_2 backend.
- In this PR, we modify the input to pass the test.

TODO: write a test to replicate the issues when passing in input_ids and attention_mask to transformers library

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)

### CODE REVIEW

File: tests/special_distributed/test_fsdp_ckpt.py [hunk 1]
```diff
@@ -26,7 +26,22 @@
...
+
+    input_ids = torch.randint(0, vocab_size, (batch_size, seq_len), device="cuda")
+
+    attention_mask = create_random_mask(
+        input_ids, max_ratio_of_left_padding=0.1, min_ratio_of_valid_token=0.5, max_ratio_of_valid_token=0.7
+    )
+    position_ids = compute_position_id_with_mask(attention_mask)
+
+    input_ids = unpad_input(input_ids.unsqueeze(-1), attention_mask)[0].transpose(0, 1)
+    position_ids = unpad_input(position_ids.unsqueeze(-1), attention_mask)[0].transpose(0, 1)
```
[the remark this answers is not in the document]
Comment (vermouth1992) [reply tests/special_distributed/test_fsdp_ckpt.py:43]: non-monotonic sequence is the correct behavior. Gemini is wrong
